services/api/mw2pdf/index.js

```diff
@@ -26,7 +27,15 @@ program
       )
     }
 
-    const pdfBooklet = await mediaWikiSession.makePdfBooklet(urls)
+    // Process passthrough parameters
+    let processedUrls = urls
+    if (options.passthroughParameters) {
+      const passthroughParameters = JSON.parse(options.passthroughParameters)
```

Comment:
Do we want to handle the case where the user gives us malformed JSON?

Reply:
Crashing and burning is *kinda* like handling....

Great idea -- will add that on Monday.

Reply:
Thinking about this more: I actually do think that if malformed JSON is provided it should halt.

That said, I'm adding more explicit validation in the hopes that it will produce more friendly error messages..

Reply:
This is great. As we discussed in the Zulip, we both agree that crashing and burning is the right thing to do. Making it explicit will really help!

Thanks for taking my feedback seriously!
Will

Reply:
Hijacking this comment to note that error handling is a place that the whole torque system is quite inadequate.  Luckily, like this, we make the very front door a bit better.  In this case, the only person passing these parameters is the system administrator who set the system up.

This isn't to say we shouldn't add these kinds of error checking, we should!  But it's more to call out that there's technical debt in general around malformed input processing in the whole system.  I kind of track it here: https://github.com/OpenTechStrategies/torque/issues/27